temporal-opentracing/src/main/java/io/temporal/opentracing/internal/OpenTracingWorkflowClientCallsInterceptor.java

```diff
@@ -0,0 +1,74 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.opentracing.internal;
+
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.util.GlobalTracer;
+import io.temporal.common.interceptors.WorkflowClientCallsInterceptor;
+import io.temporal.common.interceptors.WorkflowClientCallsInterceptorBase;
+import io.temporal.opentracing.OpenTracingOptions;
+import io.temporal.opentracing.SpanOperationType;
+
+public class OpenTracingWorkflowClientCallsInterceptor extends WorkflowClientCallsInterceptorBase {
+
+  private final OpenTracingOptions openTracingOptions;
+
+  public OpenTracingWorkflowClientCallsInterceptor(
+      WorkflowClientCallsInterceptor next, OpenTracingOptions openTracingOptions) {
+    super(next);
+    this.openTracingOptions = openTracingOptions;
+  }
+
+  @Override
+  public WorkflowStartOutput start(WorkflowStartInput input) {
+    createAndPassWorkflowStartSpan(
+        input, openTracingOptions.getSpanOperationNamePrefix(SpanOperationType.WORKFLOW_START));
+    return super.start(input);
```

Comment:
I think we should close this span only after the start gRPC call completes.

Reply:
It's go-sdk approach though I believe: https://github.com/temporalio/sdk-go/blob/cec615e1aaf798e1b153ba3918b4d3027edde808/internal/internal_workflow_client.go#L199

I'm not against your proposal, we can close in finally. But can you give more motivation or rasoning here? These two solutions lead to the same result for me, maybe I don't see something.

Reply:
My idea was to have a span that includes the start gRPC call latency and possibly retries. I'm not sure if it is useful.

Reply:
Alright, makes sense, let me do it this way.